luigi/tools/range.py

```diff
@@ -374,28 +374,29 @@ def _get_per_location_glob(tasks, outputs, regexes):
         if m is None:
             raise NotImplementedError("Couldn't deduce datehour representation in output path %r of task %s" % (p, t))
 
-    positions = [most_common((m.start(i), m.end(i)) for m in matches)[0] for i in range(1, 5)]  # the most common position of every group is likely to be conclusive hit or miss
+    n_groups = len(matches[0].groups())
+    positions = [most_common((m.start(i), m.end(i)) for m in matches)[0] for i in range(1, n_groups + 1)]  # the most common position of every group is likely to be conclusive hit or miss
 
-    glob = list(paths[0])  # TODO sanity check that it's the same for all paths?
+    glob = list(paths[0])  # FIXME sanity check that it's the same for all paths
```

Comment:
In my mind TODO is for something missing but not astonishingly so, while FIXME is for when a function or UX is broken or quite inevitably going to break astonishingly.

In this case I judged astonishment by my own inability to recall what I expected from this code :) Today I expected path formatting like `MockFile(self.dh.strftime(('/even' if self.dh.hour % 2 == 0 else '/odd') + '/%Y%m%d%H'))` to trigger an exception from this inference code but it didn't.